modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -1631,6 +1631,10 @@ private String getSingleSchemaType(Schema schema) {
     private String getPrimitiveType(Schema schema) {
         if (schema == null) {
             throw new RuntimeException("schema cannot be null in getPrimitiveType");
+        } else if (typeMapping.containsKey(schema.getType() + "_" + schema.getFormat())) {
+            // allows custom type_format mapping.
+            // use {type}_{format}
+            return typeMapping.get(schema.getType() + "_" + schema.getFormat());
```

Comment:
@wing328 I like the idea proposed in this PR of supporting type definitions for type+format combinations. I wonder if instead of an underscore, we may want to use something which is unlikely to exist in the wild (like `+`).

Compare:

```
generate -i openapi.json -o out -g kotlin --type-mappings integer_byte=kotlin.Byte
generate -i openapi.json -o out -g kotlin --type-mappings integer+byte=kotlin.Byte
generate -i openapi.json -o out -g kotlin --type-mappings string_secure_string=org.owasp.passfault.impl.SecureString
generate -i openapi.json -o out -g kotlin --type-mappings string+secure_string=org.owasp.passfault.impl.SecureString
```

Reply:
I think `@format:` could be another option. 

something like

`generate -i openapi.json -o out -g kotlin --type-mappings integer@format:byte=kotlin.Byte`

Reply:
I've created https://github.com/OpenAPITools/openapi-generator/pull/9285 based on this PR and use `+` instead. (sorry for delay in reviewing this excellent enhancement)